internal/services/network/web_application_firewall_policy_resource.go

```diff
@@ -607,6 +636,27 @@ func expandWebApplicationFirewallPolicyWebApplicationFirewallCustomRule(input []
 			RuleType:        webapplicationfirewallpolicies.WebApplicationFirewallRuleType(ruleType),
 		}
 
+		if rateLimitDuration := v["rate_limit_duration"].(string); rateLimitDuration != "" {
+			result.RateLimitDuration = pointer.To(webapplicationfirewallpolicies.ApplicationGatewayFirewallRateLimitDuration(rateLimitDuration))
+		}
+
+		if rateLimitThreshHold := v["rate_limit_threshold"].(int); rateLimitThreshHold > 0 {
+			result.RateLimitThreshold = pointer.To(int64(rateLimitThreshHold))
+		}
+
+		if groupBy := v["group_rate_limit_by"].(string); len(groupBy) > 0 {
```

Comment:
We should be doing an ok check on these 
```suggestion
		if rateLimitDuration, ok := v["rate_limit_duration"]; ok && rateLimitDuration.(string) != "" {
			result.RateLimitDuration = pointer.To(webapplicationfirewallpolicies.ApplicationGatewayFirewallRateLimitDuration(rateLimitDuration))
		}

		if rateLimitThreshHold, ok := v["rate_limit_threshold"]; ok && rateLimitThreshHold.(int) > 0 {
			result.RateLimitThreshold = pointer.To(int64(rateLimitThreshHold))
		}

		if groupBy, ok := v["group_rate_limit_by"].(string); ok && groupBy.(string) != "" {
```